lib/galaxy/model/store/__init__.py

```diff
@@ -639,9 +639,7 @@ def import_folder(folder_attrs, root_folder=None):
                 and not self.sessionless
                 and self.import_options.allow_edit
             ):
-                library_folder = self.sa_session.query(model.LibraryFolder).get(
-                    self.app.security.decode_id(library_attrs["id"])
-                )
+                library_folder = self.sa_session.query(model.LibraryFolder).get(library_attrs["id"])
```

Comment:
I'm surprised this is necessary, the model store shouldn't be touched directly by anything passing through the API. Was there an error that removing the decoding fixes here ?

Reply:
(If anything I could imagine having to encode the libraryfolder id in the corresponding export method, if that used to happen in serialize_for methods ...)

Reply:
Yeah, there was an integration test that fail after the decoding changes `test/integration/test_extended_metadata.py::ExtendedMetadataIntegrationTestCase::test_fetch_data_library` I was also surprised this decode was necessary before...

Reply:
OK, this is correct, it is coming in via the fetch data API.